yksmp/src/lib.rs

```diff
@@ -0,0 +1,375 @@
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use std::convert::TryInto;
+use std::ffi::c_void;
+use std::{ptr, slice};
+
+/// Parses LLVM stackmaps version 3 from a given address. Provides a way to query relevant
+/// locations given the return address of a `__llvm_deoptimize` function.
+/// Note that LLVM currently only supports stackmaps for 64 bit architectures. Once they support
+/// others we will need to either make this parser more dynamic or create a new one for each
+/// architecture.
+struct StackMapParser<'a> {
+    data: &'a [u8],
+    offset: usize,
+}
+
+struct Function {
+    addr: u64,
+    record_count: u64,
+}
+
+struct Record {
+    offset: u32,
+    locs: Vec<Location>,
+    liveouts: Vec<LiveOut>,
+}
+
+#[derive(Debug)]
+enum Location {
+    Register(u16),
+    Direct(u16, i32),
+    Indirect(u16, i32),
+    Constant(u32),
+    LargeConstant(u64),
+}
+
+#[allow(dead_code)]
```

Comment:
I'm OK with leaving these in for now, as long as we add a `FIXME: is this needed` or something.

Reply:
I think I completely changed my mind about it in the last hour. It's unlikely we'll need any of this stuff, and if we absolutely do, we can always easily put it back in. This then allows us to simplify `SMQuery`, or more accurately get rid of it entirely and just return a HashMap instead. But let's see what @ltratt thinks first.

Reply:
If we don't need 'em, let's nuke 'em.

Reply:
Fixed in bcef01a.